atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/OracleConnectionConfig.java

```diff
@@ -250,4 +253,30 @@ public interface ServiceNameConfiguration {
 
         class Builder extends ImmutableServiceNameConfiguration.Builder {}
     }
+
+    @Override
+    public String getSqlExceptionOverrideClassname() {
+        return OracleSqlExceptionOverride.class.getName();
+    }
+
+    public static class OracleSqlExceptionOverride implements SQLExceptionOverride {
+
+        // Must be public as Hikari instantiates via reflection
+        public OracleSqlExceptionOverride() {}
+
+        /*
+         * See https://github.com/brettwooldridge/HikariCP/pull/1838
+         * HikariCP introduced a "bug" in https://github.com/brettwooldridge/HikariCP/issues/1308
+         * which treats QueryTimeout as a connection close + evict event.
+         * This is not correct - SQLTimeoutException inherits from SQLTransientException
+         * and therefore Hikari should not close the connection. We have existing Gotham tests that assert this
+         * behaviour.
+         */
+        @java.lang.Override
+        public SQLExceptionOverride.Override adjudicate(SQLException sqlException) {
+            return sqlException instanceof SQLTimeoutException
+                    ? SQLExceptionOverride.Override.DO_NOT_EVICT
+                    : SQLExceptionOverride.Override.CONTINUE_EVICT;
```

Comment:
This would override the default adjudication behaviour for all exceptions - right now, Hikari seems to always evict on exception (default interface method, and seems to be the behavior in `checkException`), so this change maintains existing behaviour for exceptions other than SQLTimeoutException. 

On the other hand, if Hikari decides to not evict on some other exceptions and change the adjudication default method, then we'd start evicting where Hikari does not.

Not familiar enough with Hikari to judge whether this is likely, but it could be worth delegating to the interface's default if it's not SQLTimeoutException